File: snippets/node/create_an_event.js
const mailchimp = require("@mailchimp/mailchimp_marketing");
const md5 = require("md5");

mailchimp.setConfig({
  apiKey: "YOUR_API_KEY",
  server: "YOUR_SERVER_PREFIX"
});

const referralCode = "THE_REFERRAL_CODE";
const listId = "YOUR_LIST_ID";
const user = fakeDB.getUserByReferralCode(referralCode);
const subscriberHash = md5(user.email.toLowerCase());
const options = {
  name: "registered_referral",
  properties: {
    referralCode
  }
};

async function run() {
  const response = await mailchimp.lists.createListMemberEvent(
    listId,
    subscriberHash,
    options
  );

  console.log(`Event successfully created!`);
}

run();

File: snippets/ruby/create_an_event.rb
require "MailchimpMarketing"
require "digest/md5"

mailchimp = MailchimpMarketing::Client.new
mailchimp.set_config({
  :api_key => ENV["API_KEY"],
  :server => ENV["API_SERVER"]
})

referral_code = "THE_REFERRAL_CODE"
list_id = "YOUR_LIST_ID"
user = fakeDB.get_user_by_referral_code(referral_code)
subscriber_hash = Digest::MD5.hexdigest(user.email.downcase)

options = {
  name: "registered_referral",
  properties: {
    referral_code: referral_code
  }
}

response = mailchimp.lists.create_list_member_event(
  list_id,
  subscriber_hash,
  options
)

puts "Event successfully created!"